tools/test-backend

```diff
@@ -123,7 +123,7 @@ def block_internet():
                         "https://zulip.readthedocs.io/en/latest/testing.html#internet-access-inside-test-suits")
 
     httplib2.Http.request = internet_guard
-    requests.request = internet_guard
+    requests.request = lambda method, url, **kwargs: internet_guard(method, url, **kwargs)
```

Comment:
Huh, interesting that this gives an error. Seems like the signature `internet_guard` already has is as general as possible, except for the return type (which doesn't change here) -- it ought to be usable where any other function that returns `None` is usable.

If `--strict-optional` causes an error here where there wasn't one already, then that's a regression in `--strict-optional` and we should pursue it upstream. In the meantime we can have a `type: ignore` here, with a comment mentioning the upstream bug.